src/ui/opengl/castleinternalsettings.pas

```diff
@@ -261,25 +289,24 @@ procedure SettingsLoad(const Container: TUIContainer; const SettingsUrl: String)
         E.AttributeSingleDef('reference_height', DefaultUIReferenceHeight);
     end;
 
-    E := SettingsDoc.DocumentElement.Child('default_font', false);
-    if E <> nil then
+    NewFonts[ftRegular] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font', false));
+    if NewFonts[ftRegular] = nil then
+      NewFonts[ftRegular] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_regular', false));
+    NewFonts[ftBold] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_bold', false));
+    NewFonts[ftItalic] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_italic', false));
+    NewFonts[ftBoldItalic] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_bolditalic', false));
+    if (NewFonts[ftBold] = nil) and (NewFonts[ftItalic] = nil) and
+      (NewFonts[ftBoldItalic] = nil) then
+      NewDefaultFont := NewFonts[ftRegular]
```

Comment:
I am unsure about this "auto-detection" (to not use TFontFamily when only "regular" is defined).

I think it's better to just always use `TFontFamily` when user defines even only "regular" font.

So, in the new XML syntax I propose in previous comment, it would mean that 

- If you use the old method, `<default_font url="..."/>` then you don't create TFontFamily.

- If you use the new method, `<default_font><regular url="..." /></default_font>` then you always create a TFontFamily (even if bold, italic, bold_italic were not provided).

This corresponds more directlly to what is possible in code: you can have TFontFamily with only TFontFamily.Regular <> nil.